nuklear.h

```diff
@@ -4339,19 +4331,6 @@ template<typename T> struct nk_alignof{struct Big {T x; char c;}; enum {
 #define NK_UINT_MIN 0
 #define NK_UINT_MAX 4294967295u
 
-/* Make sure correct type size:
- * This will fire with a negative subscript error if the type sizes
- * are set incorrectly by the compiler, and compile out if not */
-NK_STATIC_ASSERT(sizeof(nk_size) >= sizeof(void*));
-NK_STATIC_ASSERT(sizeof(nk_ptr) == sizeof(void*));
-NK_STATIC_ASSERT(sizeof(nk_flags) >= 4);
-NK_STATIC_ASSERT(sizeof(nk_rune) >= 4);
-NK_STATIC_ASSERT(sizeof(nk_ushort) == 2);
-NK_STATIC_ASSERT(sizeof(nk_short) == 2);
-NK_STATIC_ASSERT(sizeof(nk_uint) == 4);
-NK_STATIC_ASSERT(sizeof(nk_int) == 4);
-NK_STATIC_ASSERT(sizeof(nk_byte) == 1);
```

Comment:
Why to remove this? It's a very important part of the library. We shall **not** remove this at any cost.

Reply:
Because it's redundant? It's defined in the header.

Reply:
It is for cases when `nuklear.h` is included two times once as header and later again inside a source file for implementation. In these cases both `include`s require the same defines. So to make sure that both times we have correct type sizes I added to compile time asserts once for the header and once for the implementation.

Reply:
@vurtun Correct me if I'm wrong, but if `nuklear.h` is included at least one, anywhere in the translation unit, be it directly in the source or indirectly through the `#include` chain, the static assertions are going to be there because they're in the API header part, and not the implementation part. To put it in other words, you can include the header without the implementation, but not the other way around (unless you define `NK_NUKLEAR_H_` before the `#include`, but then it won't even compile because of missing definitions).

So I just can't see any way to include the implementation without the API, but maybe I'm missing something.